components/gallery/GalleryItemToolBar.vue

```diff
@@ -0,0 +1,113 @@
+<template>
+  <div class="toolbar">
+    <NeoTooltip :label="$t('reload')" position="top">
+      <a no-shadow @click="handleReloadClick">
+        <NeoIcon
+          :icon="isLoading ? 'spinner-third' : 'arrow-rotate-left'"
+          size="medium"
+          :label="$t('reload')"
+          :spin="isLoading" />
+      </a>
+    </NeoTooltip>
+    <NeoTooltip :label="$t('fullscreen')" position="top">
+      <a no-shadow @click="$emit('toggle')">
+        <NeoIcon
+          icon="arrow-up-right-and-arrow-down-left-from-center"
+          size="medium" />
+      </a>
+    </NeoTooltip>
+    <NeoTooltip :label="$t('newTab')" position="top">
+      <a no-shadow @click="handleNewTab">
+        <NeoIcon icon="arrow-up-right" size="medium" />
+      </a>
+    </NeoTooltip>
+  </div>
+</template>
+
+<script setup lang="ts">
+import { NeoIcon, NeoTooltip } from '@kodadot1/brick'
+
+import { useGalleryItem } from './useGalleryItem'
+import { MediaType } from '@/components/rmrk/types'
+import { resolveMedia } from '@/utils/gallery/media'
+
+type ReloadElement =
+  | HTMLIFrameElement
+  | HTMLVideoElement
+  | HTMLImageElement
+  | null
+
+const isLoading = ref(false)
+defineEmits(['toggle'])
+
+const galleryItem = useGalleryItem()
+
+const { nftAnimationMimeType, nftMimeType } = galleryItem
+
+function handleReloadClick() {
+  isLoading.value = true
+
+  const reloadElement = (selector: string, isImage?: boolean) => {
+    setTimeout(() => {
+      isLoading.value = false
+      const element: ReloadElement = document.querySelector(selector)
+      if (element) {
+        if (isImage) {
+          const timestamp = new Date().getTime()
+          const url = new URL(element.src)
+          url.searchParams.set('t', timestamp.toString())
+          element.src = url.toString()
+        } else {
+          element.src += ''
+        }
+      }
+    }, 500)
+  }
+
+  const mediaType = resolveMedia(nftAnimationMimeType.value)
+  const imageType = resolveMedia(nftMimeType.value)
```

Comment:
since these do not change , no need to recalculate them every time,  move them inside a computed